lib/services/cups.pm

```diff
@@ -0,0 +1,106 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Package for cups service tests
+#
+# Maintainer: Lemon Li <leli@suse.com>
+
+package services::cups;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use Utils::Systemd 'disable_and_stop_service';
+use strict;
+use warnings;
+
+sub install_service {
+    zypper_call("in cups", exitcode => [0, 102, 103]);
+}
+
+sub config_service {
+    script_run 'echo FileDevice Yes >> /etc/cups/cups-files.conf';
+    validate_script_output 'cupsd -t', sub { m/is OK/ };
+}
+
+sub enable_service {
+    systemctl 'enable cups.service';
+}
+
+sub start_service {
+    systemctl 'start cups.service';
+}
+
+# check service is running and enabled
+sub check_service {
+    systemctl 'is-enabled cups.service';
+    systemctl 'is-active cups';
+}
+
+# check cups function
+sub check_function {
```

Comment:
Wouldn't be better to reuse existing code in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/cups.pm instead of copy pasting the whole test?

Reply:
In fact, I'm updating code of tests/console/cups.pm to reuse the functions of lib/services/cups.pm. Thanks, I will update it later.